File: 이건희/level2/짝지어 제거하기.py
def solution(s):
    num_list = []

    for i in s:
        if len(num_list) == 0:
            num_list.append(i)
        else:
            if num_list[-1] == i:
                num_list.pop()
            else:
                num_list.append(i)

    if len(num_list) == 0:
        return 1
    else:
        return 0

File: 이훈일/Programmers/Level2/짝지어 제거하기.java
import java.util.Stack;

class Solution{
    public int solution(String s){
        Stack<String> stack = new Stack<>();

        String[] sArr = s.split("");
        for(String str : sArr) {
            if(stack.empty()) {
                stack.push(str);
            }else {
                if(stack.peek().equals(str)) {
                    stack.pop();
                }else {
                    stack.push(str);
                }
            }
        }

        return stack.size()==0?1:0;
    }
}
